parlai/crowdsourcing/tasks/wizard_of_internet/acceptability.py

```diff
@@ -0,0 +1,237 @@
+        if not selections or selections[0][0]:
+            continue
+
+        search_query = msg['task_data']['search_query']
+        search_terms.append(search_query)
+        if message_history_tokens.isdisjoint(tokenize_text(search_query, stemmer)):
+            n_search_query_not_in_history += 1
+
+        selected_parts = []
+        for doc_id in range(1, len(selections)):
+            doc_selections = selections[doc_id]
+            for sentence_id in range(len(doc_selections)):
+                if doc_selections[sentence_id]:
+                    selected_parts.append(
+                        msg['task_data']['text_candidates'][doc_id - 1]['content'][
+                            sentence_id
+                        ]
+                    )
+
+        selected_knowledge.append(
+            {'text': msg['text'], 'knowledge': ' '.join(selected_parts)}
+        )
+
+    knowledge_length = []
+    knowledge_overlaps = []
+    for knwldg in selected_knowledge:
+        knowledge_tokens = tokenize_text(knwldg['knowledge'], stemmer)
+        knowledge_length.append(len(knowledge_tokens))
+
+        response_tokens = tokenize_text(knwldg['text'], stemmer)
+        knowledge_overlaps.append(overlap_ratios(knowledge_tokens, response_tokens))
+
+    violations = []
+
+    # Repeated the same search queries
+    if len(search_terms) - len(set(search_terms)) > 3:
+        violations.append(POOR_SEARCH_QUERIES)
+
+    # Search doesn't have overlap with message history
+    if n_search_query_not_in_history > 2:
+        violations.append(IRRELEVANT_SEARCH__QUERIES)
+
+    # No selection
+    if not knowledge_length:
+        violations.append(NOT_ENOUGH_SEARCH)
+
+    # Only selecting short sentences
+    if np.average(knowledge_length) < 5:
+        violations.append(SELECTED_SHORT_PIECES)
+
+    # Small overlap between response and the selected knowledge parts
+    knowledge_overlap_avg = np.average(knowledge_overlaps)
+    if knowledge_overlap_avg < knwldg_ovlp_thrshld:
+        violations.append(f'{LOW_KNOWLEDGE_OVERLAP} ({knowledge_overlap_avg})')
+
+    return violations
+
+
+class WizardOfInternetAcceptabilityChecker(AcceptabilityChecker):
+    """
+    ParlAI general acceptabilty checker customized for the wizard of internet.
+    """
+
+    def __init__(self):
+        self.knowledge_overlap_threshold = DEFAULT_KNOWLEDGE_OVERLAP_THRESHOLD
+        super().__init__()
+
+    def check_messages(
+        self,
+        agent_id: str,
+        persona: str,
+        messages: List[str],
+        is_worker_0: bool,
+        violation_types: Iterable[str] = (),
+    ) -> str:
+        violations = []
+        general_chat_violations = super().check_messages(
+            self.get_conversation_messages(messages, agent_id),
+            is_worker_0,
+            violation_types,
+        )
+        if general_chat_violations:
+            violations.extend(general_chat_violations.split(','))
+
+        ps = PorterStemmer()
```

Comment:
nit: is it necessary to instantiate this each call? or could we just do so in `__init__`?

Reply:
You are right. It is better to do it like that. Changed it.